api/agent/lb_agent.go

```diff
@@ -177,8 +185,41 @@ func (a *lbAgent) Submit(callI Call) error {
 	statsDequeue(ctx)
 	statsStartRun(ctx)
 
-	err = a.placer.PlaceCall(a.rp, ctx, call)
-	return a.handleCallEnd(ctx, call, err, true)
+	errPlace := make(chan error, 1)
+
+	call.w = &AckSyncResponseWriter{
+		origin: call.ResponseWriter(),
+		acked:  make(chan struct{}, 1),
+	}
+	isAckSync := call.Type == models.TypeAcksync
+	rw := call.w.(*AckSyncResponseWriter)
+	// change the context if it is a acksync call
+	if isAckSync {
+		ctx = common.BackgroundContext(ctx)
+		// We don't want this to run indefinetely we need to guard this context
+		var cancel func()
+		ctx, cancel = context.WithTimeout(ctx, time.Duration(60+call.Timeout))
+		defer cancel()
```

Comment:
this'll fire after Submit() exists and it'll cancel the spawned go routine below.